### PR TITLE
enhance log_file_format to fix problem when %(name)s template value is used for log directory

### LINKED ISSUE #2815: Can't use string templates such as %(name)s in the DIR part of --logfile-format
If I use e.g.

```
--logfile-format="easybuild-%(name)s,%(name)s-%(version)s-%(date)s.%(time)s.ebout.log"
```

However, the %(name)s in the directory-part is not replaced. So, I get folders that are actually called `'easybuild-%(name)s'`.

The documentation claims that these string templates should be usable for both the filename and directory: 

> The logfile format configuration setting contains a tuple specifying a log directory name and a template log file name. In both of these values, using the following string templates is supported:

See: https://easybuild.readthedocs.io/en/latest/Configuration.html#logfile-format

### PR DESCRIPTION
**Pull request by boegel**
fixes #2815

### CODE REVIEW

File: test/framework/config.py [hunk 1]
```diff
@@ -570,6 +572,89 @@ def test_strict(self):
...
+        # default
+        self.assertEqual(log_path(), 'easybuild')
+
+        # providing template values doesn't affect the default
+        ec = {'name': 'foo', 'version': '1.2.3'}
+        res = log_path(ec=ec)
+        self.assertEqual(res, 'easybuild')
+
+        # reconfigure with value for log directory that includes templates
+        init_config(args=['--logfile-format=easybuild-%(name)s-%(version)s-%(date)s-%(time)s,log,txt'])
```
Comment (ocaisa) [inline test/framework/config.py]: I don't think this option should be valid, isn't your log file going to be `log,txt` in this case?

Comment (ocaisa) [reply test/framework/config.py]: Nope, just `log`, everything from the last comma is ignored

Comment (boegel) [reply test/framework/config.py]: Oops, yeah. It'll probably be `log` rather than `log,txt`, but you're right, this doesn't make sense...

Comment (boegel) [reply test/framework/config.py]: fixed in 884413b1c

**Comment by boegel** [comment]
kicking Travis now #2819 is merged which fixes the problems with the bootstrap script...

**Pull request merged**